internal/resources/oncall/resource_user_notification_rule.go

```diff
@@ -0,0 +1,328 @@
+package oncall
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	onCallAPI "github.com/grafana/amixr-api-go-client"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	resourceUserNotificationRuleName = "grafana_oncall_user_notification_rule"
+
+	userNotificationRuleTypeOptions = []string{
```

Comment:
should MS Teams be in this list?

Reply:
https://github.com/grafana/oncall/pull/4628/files#diff-7225c8889a4ef0527f6d724edcb8482d7f3f89e41e8e6e68e393214d33f8aae1R38